library/cartridge_failover_promote.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+
+from ansible.module_utils.helpers import Helpers as helpers
+
+argument_spec = {
+    'console_sock': {'required': True, 'type': 'str'},
+    'failover_promote_params': {'required': False, 'type': 'dict'},
+}
+
+
+def failover_promote(params):
+    console_sock = params['console_sock']
+
+    failover_promote_params = params.get('failover_promote_params')
+    if failover_promote_params is None:
+        return helpers.ModuleRes(changed=False)
+
+    specified_replicaset_leaders = failover_promote_params.get('replicaset_leaders')
+    force_inconsistency = failover_promote_params.get('force_inconsistency')
+
+    if not specified_replicaset_leaders:
+        return helpers.ModuleRes(changed=False)
+
+    control_console = helpers.get_control_console(console_sock)
+
+    cluster_instances = helpers.get_cluster_instances(control_console)
+    cluster_replicasets = helpers.get_cluster_replicasets(control_console)
+
+    failover_mode, _ = control_console.eval_res_err('''
+        local cartridge = require('cartridge')
+        if cartridge.failover_get_params == nil then
+            return nil
+        end
+
+        local failover_params = cartridge.failover_get_params()
+        return failover_params.mode
+    ''')
+
+    if failover_mode != 'stateful':
+        return helpers.ModuleRes(
+            failed=True,
+            msg="Leaders promotion is possible only when stateful failover is enabled"
+        )
+
+    replicaset_leaders = {}
+    for replicaset_alias, leader_alias in specified_replicaset_leaders.items():
+        cluster_replicaset = cluster_replicasets.get(replicaset_alias)
+        if cluster_replicaset is None:
+            return helpers.ModuleRes(
+                failed=True,
+                msg="Replicaset '%s' isn't found in cluster" % replicaset_alias
+            )
+
+        if leader_alias not in cluster_replicaset['instances']:
+            return helpers.ModuleRes(
+                failed=True,
+                msg="Instance '%s' isn't found in replicaset '%s'" % (leader_alias, replicaset_alias)
+            )
+
+        leader_instance = cluster_instances.get(leader_alias)
+        if leader_instance is None:
+            return helpers.ModuleRes(
+                failed=True,
+                msg="Instance '%s' isn't found in cluster" % leader_alias
+            )
+
+        if not leader_instance.get('uuid'):
+            return helpers.ModuleRes(
+                failed=True,
+                msg="Instance '%s' has no UUID" % leader_alias
+            )
+
+        replicaset_leaders.update({
+            cluster_replicaset['uuid']: leader_instance['uuid'],
```

Comment:
Is it possible that replicaset has no UUID?

Reply:
No.